### PR TITLE
feat(format): number formatting supports custom thousands separator

### PR DESCRIPTION
**Pull request by fynnfeldpausch**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

The `formatNumber` function does not support custom thousands separator.

## What is the new behavior?

The `formatNumber` function now does support custom thousands separators as last argument. (Example: German uses `,`as decimal separator and `.` as thousands separator)

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

## Other information

While at it, I translated the formatting tests and added the formatNumber to the docs.

### CODE REVIEW

File: projects/core/utils/format/pluralize.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import {TuiPluralize} from '@taiga-ui/core/types';
 
 /**
- * Выбирает корректную форму множественного числа для отображения количества
- * @param value число
- * @param args массив из трёх форм множественного числа, например ['год', 'года', 'лет']
+ * Selects the correct plural form to display.
```
Comment (MarsiBarsi) [inline projects/core/utils/format/pluralize.ts:4]: Let's deprecate it and point out that this thing is for Russian language (and they can use an official solution from Angular https://angular.io/api/common/NgPlural)

**Review by MarsiBarsi — COMMENTED** [review]
Cool, thank you! Just a one small comment

File: projects/core/utils/format/pluralize.ts (hunk 1 above)
Comment (fynnfeldpausch) [reply projects/core/utils/format/pluralize.ts:4]: ✅

File: projects/core/utils/format/test/capitalize.spec.ts [hunk 2]
```diff
@@ -2,39 +2,37 @@ import {CHAR_NO_BREAK_SPACE} from '@taiga-ui/cdk';
 import {capitalize} from '../capitalize';
 
 describe('Capitalize', () => {
-    it('Делает первую букву слова заглавной', () => {
-        expect(capitalize('бэн')).toBe('Бэн');
```
Comment (waterplea) [inline projects/core/utils/format/test/capitalize.spec.ts:6]: R.I.P. Full Throttle references 😥

**Review by waterplea — APPROVED** [review]

File: projects/core/utils/format/pluralize.ts (hunk 1 above)
Comment (Mistydjones007) [reply projects/core/utils/format/pluralize.ts:4]: > Let's deprecate it and point out that this thing is for Russian language (and they can use an official solution from Angular https://angular.io/api/common/NgPlural)

Comment (Mistydjones007) [reply projects/core/utils/format/pluralize.ts:4]: > Let's deprecate it and point out that this thing is for Russian language (and they can use an official solution from Angular https://angular.io/api/common/NgPlural)

**Review by Mistydjones007 — APPROVED** [review]

File: projects/core/utils/format/test/capitalize.spec.ts (hunk 2 above)
Comment (fynnfeldpausch) [reply projects/core/utils/format/test/capitalize.spec.ts:6]: Ok, I did not get that one :D

**Review by MarsiBarsi — APPROVED** [review]

**Pull request merged**